hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -291,15 +302,15 @@ public void bootstrap(Option<Map<String, String>> extraMetadata) {
     if (config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
       throw new HoodieException("Cannot bootstrap the table in multi-writer mode");
     }
-    HoodieTable<T, I, K, O> table = getTableAndInitCtx(WriteOperationType.UPSERT, HoodieTimeline.METADATA_BOOTSTRAP_INSTANT_TS);
+    HoodieTable<T, I, K, O> table = initTable(WriteOperationType.UPSERT, Option.ofNullable(HoodieTimeline.METADATA_BOOTSTRAP_INSTANT_TS));
     rollbackFailedBootstrap();
     table.bootstrap(context, extraMetadata);
   }
 
   /**
    * Main API to rollback failed bootstrap.
    */
-  public void rollbackFailedBootstrap() {
```

Comment:
why moving it from public to protected? we can leave it as is. I don't think anyone outside might be having their own write client and implement these public methods. but curious on why changing it now ?

Reply:
Client is publicly exposed API, we need to be extremely careful what we expose as public -- it essentially has a chance to become public API that we'll be on the hook for to maintain (semver-wise)